### DISCUSSION TITLE
PM991 NVMe 128GB recognize only one time and Not...

### DISCUSSION THREAD

**Discussion by sangjela** [Joshua-Riek/ubuntu-rockchip#141] [Q&A]
Thanks as always for your efforts. 

I successfully booted to v1.13(At this point, the language/keyboard selection to english and add language Korean)., 
then installed nvme, booted, and saw nvme added. (on desktop, and sudo lsblk work well)
(This nvme was previously formatted in windows 11)

Im Happly prepare step to run https://github.com/Joshua-Riek/ubuntu-orange-pi5/discussions/66 (nvme boot prepare)

Before doing the spi flash, I wanted to upgrade to ubuntu-orange-pi5 v1.14, so I wrote it to a new sd card and set it up.
(At this point, the language/keyboard selection to Korean).
However, from this point on, NVME is not recognized. 

I did the following tests, but they all failed to recognize NVME.  
. Prepared ubuntu-orange-pi5 v1.14 with etcher again, set to english -> failed
. Prepare Orange Pi OS arch with etcher again, set to english -> failed
. Again with etcher on another SD card, prepare ubuntu-orange-pi5 v1.14, set to english -> failed

What to check next.
Any advice would be appreciated if you think I missed something.

**Comment by ironiris** [comment]
한국인이라고 생각하고 한국어로 답변드립니다.

최종 목적은 NVMe에 우분투를 설치하는 것이지요?
그렇다면 SDCARD로 부팅을 완료한후 우분투 이미지를 SDCARD에 다운로드 합니다.
그후 
`sudo xz -dc ubuntu-22.04.2-preinstalled-desktop-arm64-orangepi5.img.xz|dd of=/dev/nvme0n1 status=progress bs=32M`
이렇게 dd 명령을 이용해서 NVMe에 직접 설치합니다.
그 다음에
`sudo dd if=/lib/u-boot-orangepi-rk3588/rkspi_loader.img of=/dev/mtdblock0 conv=notrunc`
이 명령으로 NVMe에서 부팅할수 있도록 부트로더를 설치합니다.
이후 오렌지파이를 종료하고, SDCARD를 제거한후 부팅하면 원하시는 결과를 얻으실수 있을 것입니다.

**Comment by sangjela** [comment]
@ironiris 

thanks for your reply, but my problem is not recognized nvme. 

You suggest copy sd card to nvme , but before that, my nvme is not recognized now, so i cannot do that. 
(only one time recoginzed and never after...)

have a nice day.

**Comment by sangjela** [comment]
v1.15 , nvme recognized (but dongle wifi gone. may be I must wait next update )
nvme Samsung PM991 128G and Hynix BC511 256GB success

**Reply by Joshua-Riek** [reply]
Glad to hear the nvme is found! v1.15 is a new kernel that has missing wifi drivers, I will be releasing an update tonight or tomorrow to the kernel that will have support for rtl8811cu and rtl8821cu chipsets,

**Reply by sangjela** [reply]
Thanks so much!
